sdk/storage/azure-storage-files-shares/src/share_responses.cpp

```diff
@@ -60,4 +61,115 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
     };
   }
 
+  void ListSharesPagedResponse::OnNextPage(const Azure::Core::Context& context)
```

Comment:
I'm pretty sure that if you the coding pattern I suggest, you'll simplify ALL this code dramatically and not need friends which violate the abstractions.

Reply:
Yeah, I'll try that tomorrow.